stepper/stepper_test.go

```diff
@@ -0,0 +1,95 @@
+package stepper
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+	"time"
+)
+
+// TestNewStep tests the creation of a new step.
+func TestNewStep(t *testing.T) {
+	var output bytes.Buffer
+	step := New(&output, "Test Step")
+
+	if step.GetName() != "Test Step" {
+		t.Errorf("expected step name to be 'Test Step', got '%s'", step.GetName())
+	}
+
+	if step.done != 0 {
+		t.Errorf("expected step done to be 0, got %d", step.done)
+	}
+}
+
+// TestStepCompleteSuccess tests the successful completion of a step.
+func TestStepCompleteSuccess(t *testing.T) {
+	var output bytes.Buffer
+	step := New(&output, "Test Step")
+
+	err := step.Complete(nil)
+	if err != nil {
+		t.Errorf("expected no error, got %v", err)
+	}
+
+	if step.done != 1 {
+		t.Errorf("expected step done to be 1, got %d", step.done)
+	}
+
+	expectedOutput := "\r✅ Test Step\n"
+	if output.String() != expectedOutput {
+		t.Errorf("expected output '%s', got '%s'", expectedOutput, output.String())
+	}
+}
+
+// TestStepCompleteFailure tests the failure completion of a step.
+func TestStepCompleteFailure(t *testing.T) {
+	var output bytes.Buffer
+	step := New(&output, "Test Step")
+
+	err := step.Complete(errors.New("test error"))
+	if err == nil {
+		t.Errorf("expected error, got nil")
+	}
+
+	if step.done != 1 {
```

Comment:
What does this stand for?

Reply:
Yeah! It's a way to increase the call count to `Complete()`. Judging by this though, `done` itself might not be race-condition friendly. Will have to either guard it or use an `atomic` integer.

Reply:
Updated to use an atomic counter of uint32 size.